dtests/src/test/java/io/snappydata/hydra/cluster/SnappyTest.java

```diff
@@ -0,0 +1,756 @@
+    }
+
+    protected String getDataLocation() {
+        String scriptPath = null;
+        try {
+            hd = getHostDescription();
+            scriptPath = hd.getGemFireHome() + sep + ".." + sep + "snappy"
+                    + sep + "quickstart" + sep + "data";
+        } catch (Exception e) {
+            throw e;
+        }
+        return scriptPath;
+    }
+
+    protected HostDescription getHostDescription() {
+        hd = TestConfig.getInstance()
+                .getClientDescription(RemoteTestModule.getMyClientName())
+                .getVmDescription().getHostDescription();
+        sep = hd.getFileSep();
+        return hd;
+    }
+
+    protected String getScriptLocation(String scriptName) {
+        String scriptPath = null;
+        try {
+            hd = getHostDescription();
+            scriptPath = hd.getGemFireHome() + sep + ".." + sep + "snappy"
+                    + sep + "sbin" + sep + scriptName;
+        } catch (Exception e) {
+            throw new TestException("Following error occurred while executing " + e.getMessage());
+        }
+        return scriptPath;
+    }
+
+    protected String getSnappyPath() {
+        hd = getHostDescription();
+        snappyPath = hd.getGemFireHome() + sep + ".." + sep + "snappy";
+        return snappyPath;
+    }
+
+    protected String getParamPath(String paramName) {
+        try {
+            String fullPath = null;
+            if (paramName.contains("/")) {
+                return paramName;
+            } else {
+                fullPath = snappyTest.getDataLocation() + sep + paramName;
+                File path = new File(fullPath);
+                if (path.exists()) {
+                    Log.getLogWriter().info("Data File path found under quickstart data location.");
+                } else {
+                    fullPath = snappyTest.getUserDataLocation(paramName);
+                    Log.getLogWriter().info("Data file path found under dtests data location.");
+                }
+            }
+            return fullPath;
+        } catch (Exception e) {
+            throw new TestException("Following error occurred while executing " + e.getMessage());
+        }
+
+    }
+
+    protected String getUserScriptLocation(String scriptName) {
+        String scriptPath = null;
+        try {
+            scriptPath = hd.getGemFireHome() + sep + ".." + sep + ".." + sep + ".." + sep + "dtests" + sep + "src" + sep + "resources" + sep + "scripts" + sep + scriptName;
+        } catch (Exception e) {
+            throw new TestException("Following error occurred while executing " + e.getMessage());
+        }
+        return scriptPath;
+    }
+
+    protected String getUserDataLocation(String scriptName) {
+        String scriptPath = null;
+        try {
+            scriptPath = hd.getGemFireHome() + sep + ".." + sep + ".." + sep + ".." + sep + "dtests" + sep + "src" + sep + "resources" + sep + "data" + sep + scriptName;
```

Comment:
probably can be replaced with 
getResourcePath() {  returns hd.getGemFireHome() + sep + ".." + sep + ".." + sep + ".." + sep + "dtests" + sep + "src" + sep + "resources"  + sep 
}
dataLocation = getResourcePath() + "data" + sep + scriptName
scriptLocation = getResourcePath() +  "scripts" + sep + scriptName;